pkg/driver/videotest/dummy.go

```diff
@@ -63,9 +61,47 @@ func (d *dummy) VideoRecord(p prop.Media) (video.Reader, error) {
 		{41, 240, 110},
 	}
 
-	if p.FrameRate == 0 {
-		p.FrameRate = 30
+	yi := p.Width * p.Height
+	ci := yi / 2
+	yy := make([]byte, yi)
+	cb := make([]byte, ci)
+	cr := make([]byte, ci)
+	yyBase := make([]byte, yi)
+	cbBase := make([]byte, ci)
+	crBase := make([]byte, ci)
+	hColorBarEnd := p.Height * 3 / 4
+	wGradationEnd := p.Width * 5 / 7
+	for y := 0; y < hColorBarEnd; y++ {
+		yi := p.Width * y
+		ci := p.Width * y / 2
+		if y > hColorBarEnd {
+		} else {
+			// Color bar
+			for x := 0; x < p.Width; x++ {
+				c := x * 7 / p.Width
+				yyBase[yi+x] = uint8(uint16(colors[c][0]) * 75 / 100)
+				cbBase[ci+x/2] = colors[c][1]
+				crBase[ci+x/2] = colors[c][2]
```

Comment:
I like it. It definitely looks better than having an if statement for just handling Chroma subsampling